elasticsearch-model/lib/elasticsearch/model/indexing.rb

```diff
@@ -153,7 +153,24 @@ def mapping(options={}, &block)
         #
         #     # => {:index=>{:number_of_shards=>1}}
         #
+        # You can specify a YAML file with settings
+        #
+        # @example Define index settings from YAML file
+        #
+        #     # config/elasticsearch/articles.yml:
+        #     #
+        #     # index:
+        #     #   number_of_shards: 1
+        #     #
+        #
+        #     Article.settings "config/elasticsearch/articles.yml"
+        #
+        #     Article.settings.to_hash
+        #
+        #     # => { "index" => { "number_of_shards" => 1 } }
+        #
         def settings(settings={}, &block)
+          settings = YAML.load_file(settings) if settings.is_a?(String)
```

Comment:
It might be a good idea to wrap `YAML.load_file(settings)` with `HashWithIndifferentAccess.new` in order to be able to still use symbols as keys.